reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/OnErrorResumeNext.kt

```diff
@@ -0,0 +1,37 @@
+package com.badoo.reaktive.completable
+
+import com.badoo.reaktive.base.CompleteCallback
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.base.tryCatch
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+fun Completable.onErrorResumeNext(nextSupplier: (Throwable) -> Completable): Completable =
+    completableUnsafe { observer ->
+        val disposableWrapper = DisposableWrapper()
+        observer.onSubscribe(disposableWrapper)
+
+        subscribeSafe(
+            object : CompletableObserver, CompleteCallback by observer {
+
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onError(error: Throwable) {
+                    observer.tryCatch({ nextSupplier(error) }) {
```

Comment:
If `nextSupplier` throws an exception then there are two exceptions: `error` and the new one. You should compose them into `CompositeException`. Check out `doOnBeforeError` operator for usage example. Same for other similar places. Also please fix this for `Observable.onErrorResumeNext`, looks like I missed when I was reviewing that.

Reply:
Thanks for the reference to `doOnBeforeError`. Now both exceptions will be wrapped into `CompositeException` when supplier call fails.
Also please take a look at `TestObserver.assertError` with predicate